src/deluge/modulation/automation/auto_param.cpp

```diff
@@ -111,7 +111,8 @@ void AutoParam::setCurrentValueInResponseToUserInput(int32_t value, ModelStackWi
 	if (isPlaying) {
 
 		// If recording...
-		if (playbackHandler.recording != RecordingMode::OFF && modelStack->timelineCounterIsSet()) {
+		if (playbackHandler.recording != RecordingMode::OFF && modelStack->timelineCounterIsSet()
+		    && modelStack->getTimelineCounter()->armedForRecording) {
```

Comment:
could we downcast to `Clip` here instead of moving the arm state to timeline counter? I'm not sure which is better really

Reply:
I think the plus side of this approach is that it applies to song automation too if someone adds a way to toggle it later - then you could tweak away without worrying about record being on/off